src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -77,18 +78,26 @@ class ActionPlanService(
     val submittedActionPlan = updateDraftActionPlanAsSubmitted(draftActionPlan, submittedByUser)
     actionPlanValidator.validateSubmittedActionPlan(submittedActionPlan)
 
-    actionPlanSessionsService.createUnscheduledSessionsForActionPlan(submittedActionPlan)
-
     val savedSubmittedActionPlan = actionPlanRepository.save(submittedActionPlan)
     actionPlanEventPublisher.actionPlanSubmitEvent(savedSubmittedActionPlan)
 
     return savedSubmittedActionPlan
   }
 
+  fun approveActionPlan(id: UUID, user: AuthUser): ActionPlan {
+    val actionPlan = getActionPlan(id)
+    actionPlan.approvedAt = OffsetDateTime.now()
+    actionPlan.approvedBy = authUserRepository.save(user)
+
```

Comment:
Should we check somewhere that the plan has been submitted first or do we continue to rely on the methods being called by the UI are in the right sequence?

Reply:
yeh i don't think it's nessecary - althought to enforce this we could change `getActionPlan` to only return submitted action plans, kind of like the way we look up sent referrals vs draft referrals.